src/subcommands/psa_generate_key.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Generates a key.
+//!
+//! Currently a lot of the parameters of the key generation are hardcoded because it is not clear
+//! on how it will be possible in the future to generate a key from the command line. This is
+//! currently useful for playing with the tool, and demonstrating the use of Parsec.
+//!
+//! This will generate a 2048 bits RSA key pair for signing.
+
+pub use crate::cli::ParsecToolApp;
+use crate::error::ParsecToolError;
+use crate::subcommands::common::ProviderOpts;
+use crate::subcommands::ParsecToolSubcommand;
+use parsec_client::core::interface::operations::psa_algorithm::{AsymmetricSignature, Hash};
+use parsec_client::core::interface::operations::psa_generate_key;
+use parsec_client::core::interface::operations::psa_key_attributes::{
+    Attributes, Lifetime, Policy, Type, UsageFlags,
+};
+use parsec_client::core::interface::operations::{NativeOperation, NativeResult};
+use parsec_client::core::interface::requests::ProviderID;
+use parsec_client::core::operation_client::OperationClient;
+use std::convert::TryFrom;
+use structopt::StructOpt;
+
+/// Generates a key.
+#[derive(Debug, StructOpt)]
+#[structopt(name = "generate-key")]
+pub struct PsaGenerateKeySubcommand {
+    #[structopt(short = "k", long = "key-name")]
+    key_name: String,
+
+    #[structopt(flatten)]
+    provider_opts: ProviderOpts,
+}
+
+impl TryFrom<&PsaGenerateKeySubcommand> for NativeOperation {
+    type Error = ParsecToolError;
+
+    fn try_from(
+        psa_generate_key_subcommand: &PsaGenerateKeySubcommand,
+    ) -> Result<NativeOperation, Self::Error> {
+        Ok(NativeOperation::PsaGenerateKey(
```

Comment:
Maybe it's worth adding a `// FIXME` or `// TODO` here to signify that there are currently no options for controlling which kind of key gets generated, but not a blocker.

Reply:
Good idea, done!